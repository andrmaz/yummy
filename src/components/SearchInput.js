import React from 'react';
import PropTypes from 'prop-types';
import { Controller } from 'react-hook-form';
import { Input } from '@chakra-ui/react';

const SearchInput = ({ control, errors }) => {
    return (
        <Controller
            name="input"
            data-testid="a"
            control={control}
            rules={{ required: true }}
            defaultValue=""
            render={({ onChange, value }) => 
                <Input 
                    id='input'
                    onChange={onChange} 
                    value={value} 
                    name='input'
                    pr='4.5rem'
                    type='text'
                    variant='outline'
                    placeholder='Enter an ingredient'
                    isRequired={true}
                    errorBorderColor ='red.500'
                    isInvalid={errors.input ? true : false}
                />
            }
        />
    );
};

SearchInput.propTypes = {
    control: PropTypes.object,
    errors: PropTypes.object,
};

export default SearchInput;
