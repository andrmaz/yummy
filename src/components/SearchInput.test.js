import React from 'react';
import { shallow } from 'enzyme';
import SearchInput from './SearchInput';

let wrapper;

describe('SearchInput Component', () => {
    beforeEach(() => {
        wrapper = shallow(<SearchInput />);
        expect.hasAssertions();
    });
    it('should render Component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should as value to be required', () => {
        expect(wrapper.prop('rules')).toHaveProperty('required', true);        
    });
});