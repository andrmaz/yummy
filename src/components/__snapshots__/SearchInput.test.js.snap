// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput Component should render Component correctly 1`] = `
<Controller
  data-testid="a"
  defaultValue=""
  name="input"
  render={[Function]}
  rules={
    Object {
      "required": true,
    }
  }
/>
`;
